src/main/java/duke/AddTaskCommand.java

```diff
@@ -0,0 +1,74 @@
+package main.java.duke;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+public class AddTaskCommand extends Command {
+
+    private String description;
+    private String type;
+    private String toDoBy;
+
+    /**
+     * Constructs an instance of Add Task Command.
+     *
+     * @param description Description of task.
+     * @param type Type of task, either ToDo, Deadline or Event.
+     */
+    public AddTaskCommand(String description, String type) {
+        this.description = description;
+        this.type = type;
+    }
+
+    /**
+     * Constructs an instance of Add Task Command.
+     *
+     * @param description Description of task.
+     * @param type Type of task, either ToDo, Deadline or Event.
+     * @param toDoBy Deadline which task must be completed by.
+     */
+    public AddTaskCommand(String description, String type, String toDoBy) {
+        this.description = description;
+        this.type = type;
+        this.toDoBy = toDoBy;
+    }
+
+    /**
+     * Executes the instance of Add Task Command.
+     *
+     * @param tasks Contains the task list.
+     * @param ui Deals with interaction with the user.
+     * @param storage Deals with loading tasks from the file and saving tasks in the file.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        try {
+            if (type.equals("todo")) {
+                tasks.addTask(new ToDo(description));
+            } else if (type.equals("deadline")) {
+                tasks.addTask(new Deadline(description, toDoBy));
+            } else {
+                tasks.addTask(new Event(description, toDoBy));
+            }
```

Comment:
That's good you used `AddCommand` to house ToDo, Deadline & Event tasks, that's what I did as well!
Perhaps you can consider using a `switch` statement for this part so it looks more clean? 

```suggestion
            switch (type) {
            case "todo":
                tasks.addTask(new ToDo(description));
                break; 
            case "deadline": 
                tasks.addTask(new Deadline(description, toDoBy));
                break;
            case "event":
                tasks.addTask(new Event(description, toDoBy));
                break;
            default:
                ui.showError("Unexpected command");
                break;
            }
```